ml_engine/online_prediction/predict.py

```diff
@@ -0,0 +1,173 @@
+    Returns:
+        A dictionary of prediction results defined by the model.
+    """
+    service = get_ml_engine_service()
+    name = 'projects/{}/models/{}'.format(project, model)
+    if version is not None:
+        name += '/versions/{}'.format(version)
+
+    response = service.projects().predict(
+        name=name,
+        body={'instances': instances}
+    ).execute()
+
+    if 'error' in response:
+        raise RuntimeError(response['error'])
+
+    return response['predictions']
+# [END predict_json]
+
+
+# [START predict_tf_records]
+def predict_tf_records(project,
+                       model,
+                       example_bytes_list,
+                       key='tfrecord',
+                       version=None):
+    """Send data instances to a deployed model for prediction
+    Args:
+        project: str, project where the Cloud ML Engine Model is deployed
+        model: str, model name.
+        example_bytes_list: [str], Serialized tf.train.Example protos.
+        version: str, version of the model to target.
+    Returns:
+        A dictionary of prediction results defined by the model.
+    """
+    import base64
+    service = get_ml_engine_service()
+    name = 'projects/{}/models/{}'.format(project, model)
+    if version is not None:
+        name += '/versions/{}'.format(version)
+
+    response = service.projects().predict(
+        name=name,
+        body={'instances': [
+            {key: {'b64': base64.b64encode(example_bytes)}}
+            for example_bytes in example_bytes_list
+        ]}
+    ).execute()
+    if 'error' in response:
+        raise RuntimeError(response['error'])
+
+    return response['predictions']
+# [END predict_tf_records]
+
+
+# [START census_to_example_bytes]
+def census_to_example_bytes(json_instance):
+    """Serialize a JSON example to the bytes of a tf.train.Example.
+    This method is specific to the signature of the Census example.
+    Args:
+        json_instance: dict, representing data to be serialized.
+    Returns:
+        A string (as a container for bytes).
+    """
+    import tensorflow as tf
+    feature_dict = {}
+    for key, data in json_instance.iteritems():
+        if isinstance(data, str) or isinstance(data, unicode):
+            feature_dict[key] = tf.train.Feature(
+                bytes_list=tf.train.BytesList(value=[str(data)]))
+        elif isinstance(data, int) or isinstance(data, float):
+            feature_dict[key] = tf.train.Feature(
+                float_list=tf.train.FloatList(value=[data]))
+    return tf.train.Example(
+        features=tf.train.Features(
+            feature=feature_dict
+        )
+    ).SerializeToString()
+# [END census_to_example_bytes]
+
+
+def main(project, model, version=None, force_tfrecord=False):
+    """Send user input to the prediction service."""
+    import json
+    while True:
+        try:
+            user_input = json.loads(raw_input("Valid JSON >>>"))
```

Comment:
It depends on their model. This snippet will be part of a docs page that is attempting to explain just that. This will be at the end "now that you know what the prediction service does, here's how you call it".